classes/components/forms/submission/ChangeSubmissionLanguageMetadataForm.php

```diff
@@ -0,0 +1,118 @@
+    {
+        $this->action = $action;
+
+        $this->addGroup([
+            'id' => 'language',
+        ]);
+
+        $submissionLocale = $submission->getData('locale');
+        $supportedLocaleNames = collect($context->getSupportedSubmissionLocaleNames());
+        $submissionLocaleNameArr = collect(Locale::getSubmissionLocaleDisplayNames([$submissionLocale]));
+        $showWhen = ['locale', $supportedLocaleNames->diffKeys($submissionLocaleNameArr)->keys()->toArray()];
+
+        $this->addGroup([
+            'id' => 'metadata',
+            'showWhen' => $showWhen,
+        ]);
+
+        // Language
+        $localeOptions = $supportedLocaleNames
+            ->union($submissionLocaleNameArr)
+            ->sortKeys()
+            ->map(fn ($name, $key) => ['value' => $key, 'label' => $name])
+            ->values()
+            ->toArray();
+        $this->addField(new FieldRadioInput('locale', [
+            'label' => __('submission.submit.submissionLocale'),
+            'description' => __('submission.list.changeSubmissionLanguage.languageDescription'),
+            'groupId' => 'language',
+            'type' => 'radio',
+            'options' => $localeOptions,
+            'isRequired' => true,
+            'value' => $submissionLocale,
+        ]));
+
+        // Metadata description
+        $this->addField(new FieldHTML('metadataDescription', [
+            'description' => __('submission.list.changeSubmissionLanguage.metadataDescription'),
+            'groupId' => 'metadata',
+        ]));
+
+        $submissionLocaleName = $submissionLocaleNameArr->get($submissionLocale);
+
+        // Title and abstract
+        $titleAbstractForm = $this->getTitleAbstractForm(FormComponent::ACTION_EMIT, [$submissionLocale], $publication, $context);
+        $this->setField($titleAbstractForm->getField('title'), $submissionLocaleName, $submissionLocale);
+        $this->setField($titleAbstractForm->getField('abstract'), $submissionLocaleName, $submissionLocale);
+
+        // Add cancel button
+        $this->addCancel();
+    }
+
+    protected function addCancel() {
+        $this->addPage([
+            'id' => 'default',
+            'submitButton' => ['label' => __('common.confirm')],
+            'cancelButton' => ['label' => __('common.cancel')],
+        ]);
+        collect($this->groups)->each(fn ($_, $i) => ($this->groups[$i]['pageId'] = 'default'));
+    }
+
+    protected function getTitleAbstractForm(string $publicationApiUrl, array $locales, Publication $publication, Context $context): TitleAbstractForm
+    {
+        $pubSecId = $publication->getData('sectionId');
+        $section = $pubSecId ? Repo::section()->get($pubSecId, $context->getId()) : null;
+        return new TitleAbstractForm(
+            $publicationApiUrl,
+            $locales,
+            $publication,
+            $section ? (int) $section->getData('wordCount') : 0,
+            $section && !$section->getData('abstractsNotRequired')
```

Comment:
If I follow correctly, you create TitleAbstractForm to cherry-pick the title and abstract fields from it? And during cherry-picking you also convert it from multilingual to not multilingual, right? Would not be easier just to add title and abstract fields directly to your form and not using the TitleAbstractForm?

Reply:
@jyhein, I am also wondering why have you chosen this way? Something specific?
It seems it is generic right now, how you have solved it, in case we need consider additional field, but... just curious...

Reply:
Currently there are only 2 fields but if more are added, then it is easier this way than repeating the code from forms. This is same if added new fields from forms, e.g. PKPMetadataForm